### PR TITLE
feat: extend asset profile dialog form

### LINKED ISSUE #2524: Extend the asset profile dialog form
Extend the [AssetProfileDialog](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.component.ts) form by `name` (required), `assetClass` and `assetSubClass` where `dataSource` is `MANUAL` (same as `scraperConfiguration`).

Here you can find examples for the [asset class selector](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/portfolio/activities/create-or-update-activity-dialog/create-or-update-activity-dialog.html#L285) and [asset sub class selector](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/portfolio/activities/create-or-update-activity-dialog/create-or-update-activity-dialog.html#L301).

extend-asset-profile-dialog-form

### PR DESCRIPTION
**Pull request by Dhoni77**
Closes #2524

- Added name (required), assetClass and assetSubClass inputs.
- Handled the form data in frontend and backend.

Attachment:

asset_dialog

### CODE REVIEW

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html [hunk 1]
```diff
@@ -112,16 +112,20 @@ <h1 class="flex-grow-1 m-0" mat-dialog-title>
         >
       </div>
       <div class="col-6 mb-3">
-        <gf-value i18n size="medium" [hidden]="!assetClass" [value]="assetClass"
+        <gf-value
+          i18n
+          size="medium"
+          [hidden]="!assetProfileClass"
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html:118]: `[hidden]="!assetProfileForm.controls['assetClass'].value"`

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html [hunk 2]
```diff
@@ -112,16 +112,20 @@ <h1 class="flex-grow-1 m-0" mat-dialog-title>
...
+          [value]="assetProfileClass"
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html:119]: `[value]="assetProfileForm.controls['assetClass'].value"`

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html [hunk 3]
```diff
@@ -112,16 +112,20 @@ <h1 class="flex-grow-1 m-0" mat-dialog-title>
...
           >Asset Class</gf-value
         >
       </div>
       <div class="col-6 mb-3">
         <gf-value
           i18n
           size="medium"
-          [hidden]="!assetSubClass"
-          [value]="assetSubClass"
+          [hidden]="!assetProfileSubClass"
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html:127]: `[hidden]="!assetProfileForm.controls['assetSubClass'].value"`

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html [hunk 4]
```diff
@@ -112,16 +112,20 @@ <h1 class="flex-grow-1 m-0" mat-dialog-title>
...
+          [value]="assetProfileSubClass"
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html:128]: `[value]="!assetProfileForm.controls['assetSubClass'].value"`

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thanks again, @Dhoni77! 😎 I have added some comments.

**Comment by Dhoni77** [comment]
cc @dtslvr

**Review by dtslvr — APPROVED** [review]
Thank you for the modifications, @Dhoni77! 👍🏻 
Ready to merge now.

**Pull request merged**